packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -739,32 +741,114 @@ export class Stack extends CoreConstruct implements ITaggable {
+      console.error(`[CDK_DEBUG_DEPS] stack "${reason.source.node.path}" depends on "${reason.target.node.path}" because: ${reason.description}`);
+    }
+  }
+
+  /**
+   * Called implicitly by the `obtainDependencies` helper function in order to
+   * collect resource dependencies across two top-level stacks at the assembly level.
+   *
+   * Use `stack.obtainDependencies` to see the dependencies between any two stacks.
+   *
+   * @internal
+   */
+  public _obtainAssemblyDependencies(reasonFilter: StackDependencyReason={ description: '' }) {
+    let dependencies: Set<Element> = new Set();
+    Object.values(this._stackDependencies).forEach((dep) => {
+      filterReasons(dep, reasonFilter).forEach(r => {
+        if (r.target) {
+          dependencies.add(r.target);
+        }
+      });
+    });
+    return Array.from(dependencies);
+  }
+
+  /**
+   * Called implicitly by the `removeDependency` helper function in order to
+   * remove a dependency between two top-level stacks at the assembly level.
+   *
+   * Use `stack.addDependency` to define the dependency between any two stacks,
+   * and take into account nested stack relationships.
+   *
+   * @internal
+   */
+  public _removeAssemblyDependency(target: Stack, reasonFilter: StackDependencyReason={ description: 'dependency removed using stack.removeDependency()' }) {
+    // defensive: we should never get here for nested stacks
+    if (this.nested || target.nested) {
+      throw new Error('There cannot be assembly-level dependencies for nested stacks');
+    }
+    // No need to check for a dependency cycle when removing one
+
+    // Fill in reason details if not provided
+    if (!reasonFilter.source) {
+      reasonFilter.source = this;
+    }
+    if (!reasonFilter.target) {
+      reasonFilter.target = target;
+    }
+
+    let dep = this._stackDependencies[Names.uniqueId(target)];
+    if (!dep) {
+      // Dependency doesn't exist - return now
+      return;
+    }
+
+    // Find and remove the specified reason from the dependency
+    let index;
+    let filteredReasons = filterReasons(dep, reasonFilter);
+    if (filteredReasons.length > 1) {
+      throw new Error(`Reason for dependency removal is too ambiguous: ${reasonFilter}`);
+    }
+    if (filteredReasons.length == 0) {
+      // Reason is already not there - return now
+      return;
+    }
+    let matchedReason = filteredReasons[0];
+
+    index = dep.reasons.indexOf(matchedReason, 0);
+    dep.reasons.splice(index, 1);
+    // If that was the last reason, remove the dependency
+    if (dep.reasons.length == 0) {
+      delete this._stackDependencies[Names.uniqueId(target)];
+    }
+
+    if (process.env.CDK_DEBUG_DEPS) {
+      // eslint-disable-next-line no-console
+      console.error(`[CDK_DEBUG_DEPS] stack "${this.node.path}" no longer depends on "${target.node.path}" because: ${reasonFilter}`);
```

Comment:
should be `throw new Error()`

Reply:
I'm not sure about this one - this is intended to be debug message for troubleshooting. Changing this to throw an exception would break this method whenever the debug flag is set. Maybe it would be more appropriate to change this to `console.log()`, since this is not an error?

Reply:
yes, this should be a log call, not an error.